src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,291 @@
+                subjectBuffer,
+                _asyncListener,
+                // Any time an edit happens, recompute as the document symbols may have changed.
+                TaggerEventSources.OnTextChanged(subjectBuffer),
+                // Switching what is the active context may change the document symbols.
+                TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer),
+                // Many workspace changes may need us to change the document symbols (like options changing, or project renaming).
+                TaggerEventSources.OnWorkspaceChanged(subjectBuffer, _asyncListener),
+                // Once we hook this buffer up to the workspace, then we can start computing the document symbols.
+                TaggerEventSources.OnWorkspaceRegistrationChanged(subjectBuffer));
+
+            _textViewEventSource.Changed += OnEventSourceChanged;
+            _textViewEventSource.Connect();
+            _codeWindowEventsSink = ComEventSink.Advise<IVsCodeWindowEvents>(codeWindow, this);
+            StartComputeDataModelTask();
+        }
+
+        public void Dispose()
+        {
+            _codeWindowEventsSink.Unadvise();
+            _textViewEventSource.Changed -= OnEventSourceChanged;
+            _textViewEventSource.Disconnect();
+            _cancellationTokenSource.Cancel();
+            _cancellationTokenSource.Dispose();
+        }
+
+        int IVsCodeWindowEvents.OnNewView(IVsTextView pView)
+        {
+            _threadingContext.ThrowIfNotOnUIThread();
+            return StartTrackingView(pView);
+        }
+
+        private int StartTrackingView(IVsTextView textView)
+        {
+            _threadingContext.ThrowIfNotOnUIThread();
+
+            var wpfTextView = _editorAdaptersFactoryService.GetWpfTextView(textView);
+            if (wpfTextView is null)
+                return VSConstants.E_FAIL;
+
+            _trackedTextViews.Add(textView, wpfTextView);
+
+            // In the split window case, there's two views (each with its own caret position) but only one text buffer.
+            // Subscribe to caret position changes once per view.
+            wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+
+            return VSConstants.S_OK;
+        }
+
+        int IVsCodeWindowEvents.OnCloseView(IVsTextView pView)
+        {
+            _threadingContext.ThrowIfNotOnUIThread();
+
+            if (_trackedTextViews.TryGetValue(pView, out var view))
+            {
+                // In the split window case, there's two views (each with its own caret position) but only one text buffer.
+                // Unsubscribe to caret position changes once per view.
+                view.Caret.PositionChanged -= Caret_PositionChanged;
+
+                _trackedTextViews.Remove(pView);
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        private void OnEventSourceChanged(object sender, TaggerEventArgs e)
+            => StartComputeDataModelTask();
```

Comment:
nit: consider renaming to Enqueue instead of Start.  'Start' implies the task will begin running immediately.  Enqueue makes it clearer it's going into a queue to run sometime in teh future.